### PR TITLE
fix: PowerShell formatting - 4 spaces tab alignment

### PR DESCRIPTION
**Pull request by eriqua**
## Description

Disable `editor.detectIndentation` vscode setting for PowerShell files to allow 4 spaces tab to be effective
Reformat all .ps1 files
> Note: only contains formatting changes

## Pipeline Reference

| Pipeline |
| -------- |
| [avm.res.network.nat-gateway](https://github.com/eriqua/bicep-registry-modules/actions/workflows/avm.res.network.nat-gateway.yml) |
| [avm.res.compute.virtual-machine-scale-set](https://github.com/eriqua/bicep-registry-modules/actions/workflows/avm.res.compute.virtual-machine-scale-set.yml) |
| [avm.res.analysis-services.server](https://github.com/eriqua/bicep-registry-modules/actions/workflows/avm.res.analysis-services.server.yml) |

## Type of Change

- [x] Update to CI Environment or utilities (Non-module affecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "editor.insertSpaces": true,
     "editor.tabSize": 4,
     "files.encoding": "utf8bom",
-    "files.insertFinalNewline": true
+    "files.insertFinalNewline": true,
+    "editor.detectIndentation": false
```
Comment (AlexanderSehr) [inline .vscode/settings.json:30]:
```suggestion
    "editor.detectIndentation": false // VS Code will not detect indentation/tab/space from the file and use settings editor.insertSpaces and editor.tabSize instead
```

**Review by AlexanderSehr — DISMISSED** [review]

**Comment by AlexanderSehr** [comment]
Hey @eriqua, just one comment/suggestion. Looks good

**Comment by eriqua** [comment]
> Hey @eriqua, just one comment/suggestion. Looks good

I don't know why but this requires finops pattern to recompile the main.json to not fail its static validation. Will move to draft until further investigation

**Comment by eriqua** [comment]
> > Hey @eriqua, just one comment/suggestion. Looks good
> 
> I don't know why but this requires finops pattern to recompile the main.json to not fail its static validation. Will move to draft until further investigation

@AlexanderSehr got it, waiting for #2498 to be merged as that is regenerating the main.json for all modules already. I believe this is due to the new Bicep version, e.g. if you look at [this diff](https://github.com/Azure/bicep-registry-modules/pull/2498/files#diff-1fc2679c622a0c21c8d688d3f207e6461de415b8b37fdbceba763e3c80ee3097) you'll see a different way `\r` and `\n` are managed

**Comment by AlexanderSehr** [comment]
> > > Hey @eriqua, just one comment/suggestion. Looks good
> > 
> > 
> > I don't know why but this requires finops pattern to recompile the main.json to not fail its static validation. Will move to draft until further investigation
> 
> @AlexanderSehr got it, waiting for #2498 to be merged as that is regenerating the main.json for all modules already. I believe this is due to the new Bicep version, e.g. if you look at [this diff](https://github.com/Azure/bicep-registry-modules/pull/2498/files#diff-1fc2679c622a0c21c8d688d3f207e6461de415b8b37fdbceba763e3c80ee3097) you'll see a different way `\r` and `\n` are managed

I wonder if we can also manage that via a VSCode setting - or if it's an issue of the utility and must be tackled there...

**Review by krbar — DISMISSED** [review]

**Review by AlexanderSehr — DISMISSED** [review]

**Review by AlexanderSehr — APPROVED** [review]

**Comment by sebassem** [comment]
> > Hey @eriqua, just one comment/suggestion. Looks good
> 
> I don't know why but this requires finops pattern to recompile the main.json to not fail its static validation. Will move to draft until further investigation

Hi @eriqua , I notice this every time pipeline runs for my modules which run PowerShell deployment scripts, usually on compile spaces in the script are translated to `\r\n` instead of `\n` so I usually have to remove the spaces and re-run it. Any thoughts on which setting might be causing this ?

**Comment by eriqua** [comment]
> > > Hey @eriqua, just one comment/suggestion. Looks good
> > 
> > 
> > I don't know why but this requires finops pattern to recompile the main.json to not fail its static validation. Will move to draft until further investigation
> 
> Hi @eriqua , I notice this every time pipeline runs for my pattern modules(finops and sub-vending) which run PowerShell deployment scripts, usually on compile spaces in the script are translated to `\r\n` instead of `\n` so I usually have to remove the spaces and re-run it. Any thoughts on which setting might be causing this ?

Hi @sebassem, not sure, I just noticed the static validation failing for both finops and sub-vending starting from Bicep version 0.28.1, independently of the latest vscode settings update. @AlexanderSehr anything from your side on this?

**Comment by AlexanderSehr** [comment]
Hey @eriqua, sub-vending looks green to me 😄 
Fin-Ops on the other hand has a apparently a `\r\n` vs `\n` issue

**Pull request merged**